client/js/lib/prosemirror/converter.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * This ProseMirrorConverter class is responsible for converting
+ * ProseMirror data to an HTML string and to convert proseMirror data
+ * to an HTML string.
+ *
+ * Example usage:
+ * - from ProseMirror to html: `converter.fromProseMirror(ProseMirrorData).toHtml().getHtml()`
+ * - from html to ProseMirror: `converter.fromHtml(htmlString).toProseMirror()`
+ *
+ * @param proseMirrorData - object
+ * @param htmlString - string
+ */
+export class ProseMirrorConverter {
+  constructor() {
+    this.prosemirrorData = null
+    this.htmlString = null
+  }
+
+  /**
+   * Returns a ProseMirrorConverter instance filled with ProseMirror data.
+   * @param proseMirrorData - string
+   * @returns { ProseMirrorConverter }
+   */
+  fromProseMirror = (proseMirrorData) => {
+    this.prosemirrorData = proseMirrorData
+    return this
+  }
+
+  /**
+   * Converts ProseMirror data to an HTML string.
+   * @returns { ProseMirrorConverter }
+   */
+  toHtml = () => {
+    try {
+      const { type, id, relationships } = this.prosemirrorData.data
+      const draftSegments = relationships.draftSegments.data.map(segment => segment.id).join(', ')
+      const statementId = relationships.statement.data.id
+
+      this.htmlString = ''
+      console.log(type, id, draftSegments, statementId)
+      // TODO: return HTML string.
+      return this
+    } catch (error) {
+      console.error('Error converting ProseMirror data to HTML: ', error)
+    }
+  }
+
+  /**
+   * Returns a ProseMirrorConverter instance filled with an HTML string.
+   * @param htmlString
+   * @returns { ProseMirrorConverter }
+   */
+  fromHtml(htmlString) {
+    try {
+      const parser = new DOMParser()
```

Comment:
I'd suggest to instanciate the parser with the class constructor so it is not instanciated everytime again, when `fromHtml` runs.

Reply:
This is a really great suggestion, thanks a lot!
Done here: https://github.com/demos-europe/demosplan-core/pull/3875/commits/269959511ba697df98843ec0a812649697c6cd31